SalesforceFoundation/EDA
# [EDA] - Primary Affiliation's lookup field API name is not found on AFFL_AccChange_TDTM, AFFL_ContactChange_TDTM and AFFL_MultiRecordType_TDTM when  'Account Record Type' &  'Contact Primary Affl Field' are both API names

# Critical Changes

# Changes
## Fix to Update the Primary Affiliation Field on Contact
We've fixed an issue that prevented the Primary Affiliation fields on Contact from updating when the Account or Contact fields on an Affiliation are updated. This fix also ensures that the Primary Affiliation fields on Contact are updated when an Affiliation is deleted that uses API names for the Account Record Type or Contact Primary Affl fields in Affiliation Mapping.

# Issues Closed
Closes #1224

# New Metadata

# Deleted Metadata

# Testing Notes
Scenario 1 (change of Account on Affiliation):

1. Create 2 education institution business accounts ‘Harvard’ and ‘MIT’
2. Create a contact 'student contact'
3. From ‘student contact’ related list create a primary affiliation to ‘Harvard’ account
4. Confirm an affiliation is created between ‘Harvard’ And ‘student contact’
5. Confirm ‘Primary Educational Institution’ on ‘student contact’ is populated with ‘Harvard’
6. Goto EDA settings -> Change ' Educational Institution' to '
    hed__Primary_Educational_Institution__c' on 'Contact Primary Affl Field' column if you are using namespaced org. If not change it to 'Primary_Educational_Institution__c' . Save.
7. .Goto the affiliation created on step 3 and change the account to 'MIT'.
8. Confirm account is changed on Affiliation (as expected)
9. *Bug*: The 'Primary Educational Institution’ on ‘student contact’ is still ‘Harvard’
10. *Expected*: The 'Primary Educational Institution’ on ‘student contact’ should be updated with 'MIT'.
    

Scenario 2 (change of Contact on Affiliation):

1. Create 2 education institution business account ‘Harvard’ and ‘MIT’
2. Create 2 contacts ‘Chad’ and ‘Brad’
3. From ‘Chad’ create a primary affiliation to ‘Harvard’ account
4. Confirm an affiliation is created between ‘Harvard’ And ‘Chad’
5. Confirm ‘Primary Educational institution’ on ‘Chad’ is populated with ‘Harvard'
6. From ‘Brad’ create a primary affiliation to ‘MIT’ account
7. Confirm an affiliation is created between ‘MIT’ And ‘Brad’
8. Confirm ‘Primary Educational institution’ on ‘Brad’ is populated with ‘MIT'
9. Goto EDA settings -> Change ' Educational Institution' to '
    hed__Primary_Educational_Institution__c' on 'Contact Primary Affl Field' column if you are using namespaced org. If not change it to 'Primary_Educational_Institution__c' . Save.
10. Now, go the primary affiliation created on step3 and change the Contact from 'Chad' to 'Brad'
11. Confirm Contact on the Affilaiton is changed to 'Brad'.
12. *Expected*: ‘Primary Educational institution’ on 'Chad' should be cleared.
13. *Bug*: ‘Primary Educational institution’ on 'Chad' is still 'Harvard'
14. *Bug*: ‘Primary Educational institution’ on 'Brad' should be 'Harvard' but is still 'MIT'

Scenario 3 (Delete the Affiliation):

1. Create an education institution business account ‘Harvard’ 
2. Create a Contact ‘Chad’
3. From ‘Chad’ create a primary affiliation to ‘Harvard’ account
4. Confirm an affiliation is created between ‘Harvard’ And ‘Chad’
5. Confirm ‘Primary Educational institution’ on ‘Chad’ is populated with ‘Harvard’
6. Delete the primary affiliation created on  step 3
7. *Expected*: ‘Primary Educational institution’ on 'Chad' should be cleared.
8. *Bug*: ‘Primary Educational institution’ on 'Chad' is still 'Harvard'
